pkg/internal/token/execCredentialPlugin.go

```diff
@@ -24,20 +23,27 @@ type ExecCredentialPlugin interface {
 type execCredentialPlugin struct {
 	o                    *Options
 	cachedRecord         CachedRecordProvider
-	popTokenCachedRecord *popcache.Cache
+	popTokenCache        *popcache.Cache
 	execCredentialWriter ExecCredentialWriter
-	newCredentialFunc    func(record azidentity.AuthenticationRecord, popCache cache.ExportReplace, o *Options) (CredentialProvider, error)
+	newCredentialFunc    func(record azidentity.AuthenticationRecord, o *Options) (CredentialProvider, error)
 }
 
 var errAuthenticateNotSupported = errors.New("authenticate is not supported")
 
 func New(o *Options) (ExecCredentialPlugin, error) {
 	klog.V(10).Info(o.ToString())
 
-	// Create PoP token cache using the official MSAL & MSAL extension libraries.
-	popTokenCache, err := popcache.NewCache(o.AuthRecordCacheDir)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create PoP token cache provider: %w", err)
+	// Initialize PoP token cache in Options if enabled
+	if o.IsPoPTokenEnabled && o.popTokenCache == nil {
+		// Create PoP token cache using the official MSAL & MSAL extension libraries.
+		popTokenCache, err := popcache.NewCache(o.AuthRecordCacheDir)
+		if err != nil {
```

Comment:
I think the logic here can move to options.go such that in the credential's constructor, we just do 
```go
popKeyProvider := o.GetPoPKeyProvider()
```
in GetToken
```go
key := popKeyProvider.GetKey() // sw key or persistent key
```